### ISSUE TITLE
Problem matchers which match an invalid file cause annotations to come up as being in the file .github

### ISSUE THREAD

**Issue by peternewman** [actions/toolkit#473]
**Describe the bug**
If a problem matcher doesn't match a valid file (e.g. the regex captures "test.txt ", with a space at the end, the annotation comes up as being in the file .github rather than the matched file.

**To Reproduce**
Create a problem matcher which matches at least an invalid filename, or possibly a file not in the repo.

**Expected behavior**
Either not annotating .github, or annotating the match for a filename that doesn't exist, or perhaps clearest some sort of error somewhere (perhaps in the action log), that the matched filename doesn't exist.

**Comment by bilderbuchi** [comment]
"annotating the match for a filename that doesn't exist" sounds most useful, and maybe if possible indicate that the file was not found in the repo. Although, tbh, `.github` is also not a file..

**Comment by peternewman** [comment]
> Although, tbh, `.github` is also not a file..

I think the issue was if it shows up on `.github` you can't tell which file it was trying to tag and therefore work out what was going on.

**Comment by bilderbuchi** [comment]
Yes, exactly. That was more a quip that `.github` is a directory, not a file. Well, it's late :-P
